diff --git a/node_modules/typedoc-plugin-versions/src/index.js b/node_modules/typedoc-plugin-versions/src/index.js
index 449898f..137a08f 100644
--- a/node_modules/typedoc-plugin-versions/src/index.js
+++ b/node_modules/typedoc-plugin-versions/src/index.js
@@ -85,7 +85,7 @@ function load(app) {
         vUtils.makeMinorVersionLinks(metadata.versions, rootPath);
         const jsVersionKeys = vUtils.makeJsKeys(metadata);
         fs_extra_1.default.writeFileSync(path_1.default.join(rootPath, 'versions.js'), jsVersionKeys);
-        fs_extra_1.default.writeFileSync(path_1.default.join(rootPath, 'index.html'), `<meta http-equiv="refresh" content="0; url=${metadata.stable ? 'stable' : 'dev'}"/>`);
+        fs_extra_1.default.writeFileSync(path_1.default.join(rootPath, 'index.html'), `<meta http-equiv="refresh" content="0; url=${metadata.stable ? 'stable/' : 'dev/'}"/>`);
         vUtils.saveMetadata(metadata, rootPath);
     });
     return vOptions;
