diff --git a/builtins/web/url.cpp b/builtins/web/url.cpp
index 6f716b6..df12f40 100644
--- a/builtins/web/url.cpp
+++ b/builtins/web/url.cpp
@@ -382,14 +382,14 @@ JSObject *URLSearchParams::create(JSContext *cx, JS::HandleObject self,
   return self;
 }

-JSObject *URLSearchParams::create(JSContext *cx, JS::HandleObject self, jsurl::JSUrl *url) {
+JSObject *URLSearchParams::create(JSContext *cx, JS::HandleObject self, JS::HandleObject url) {

-  jsurl::JSUrlSearchParams *params = jsurl::url_search_params(url);
+  jsurl::JSUrlSearchParams *params = jsurl::url_search_params(const_cast<jsurl::JSUrl*>(URL::url(url)));
   if (!params)
     return nullptr;

   JS::SetReservedSlot(self, Slots::Params, JS::PrivateValue(params));
-  JS::SetReservedSlot(self, Slots::Url, JS::PrivateValue(url));
+  JS::SetReservedSlot(self, Slots::Url, JS::ObjectValue(*url));

   return self;
 }
@@ -646,10 +646,8 @@ bool URL::searchParams_get(JSContext *cx, unsigned argc, JS::Value *vp) {
     if (!url_search_params_instance)
       return false;

-    // The const-cast here is okay because we while normally callers of URL::url mustn't mutate
-    // the returned object, URLSearchParams is intended to.
     params = URLSearchParams::create(cx, url_search_params_instance,
-                                     const_cast<jsurl::JSUrl *>(url(self)));
+                                     self);
     if (!params)
       return false;
     JS::SetReservedSlot(self, Slots::Params, JS::ObjectValue(*params));

diff --git a/builtins/web/url.h b/builtins/web/url.h
index 2fee9a3..79252e3 100644
--- a/builtins/web/url.h
+++ b/builtins/web/url.h
@@ -56,7 +56,7 @@ public:
   static jsurl::SpecSlice serialize(JSContext *cx, JS::HandleObject self);
   static jsurl::JSUrlSearchParams *get_params(JSObject *self);

-  static JSObject *create(JSContext *cx, JS::HandleObject self, jsurl::JSUrl *url);
+  static JSObject *create(JSContext *cx, JS::HandleObject self, JS::HandleObject url);
   static JSObject *create(JSContext *cx, JS::HandleObject self,
                           JS::HandleValue params_val = JS::UndefinedHandleValue);

