diff --color -ru openssl-3.0.7/crypto/uid.c openssl-3.0.7-orig/crypto/uid.c
--- a/crypto/uid.c	2023-01-09 17:00:00.094912576 -0800
+++ b/crypto/uid.c	2022-11-01 07:14:36.000000000 -0700
@@ -10,7 +10,7 @@
 #include <openssl/crypto.h>
 #include <openssl/opensslconf.h>
 
-#if defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_UEFI)
+#if defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_UEFI) || defined(__wasi__)
 
 int OPENSSL_issetugid(void)
 {
