diff --git openssl-3.0.7/crypto/rand/rand_lib.c openssl-3.0.7/crypto/rand/rand_lib.c
--- a/crypto/rand/rand_lib.c
+++ b/crypto/rand/rand_lib.c
@@ -320,6 +320,10 @@ const RAND_METHOD *RAND_get_rand_method(void)
 int RAND_priv_bytes_ex(OSSL_LIB_CTX *ctx, unsigned char *buf, size_t num,
                        unsigned int strength)
 {
+#ifdef __wasi__
+    arc4random_buf(buf, num);
+    return 1;
+#endif
     EVP_RAND_CTX *rand;
 #if !defined(OPENSSL_NO_DEPRECATED_3_0) && !defined(FIPS_MODULE)
     const RAND_METHOD *meth = RAND_get_rand_method();
@@ -349,6 +353,10 @@ int RAND_priv_bytes(unsigned char *buf, int num)
 int RAND_bytes_ex(OSSL_LIB_CTX *ctx, unsigned char *buf, size_t num,
                   unsigned int strength)
 {
+#ifdef __wasi__
+    arc4random_buf(buf, num);
+    return 1;
+#endif
     EVP_RAND_CTX *rand;
 #if !defined(OPENSSL_NO_DEPRECATED_3_0) && !defined(FIPS_MODULE)
     const RAND_METHOD *meth = RAND_get_rand_method();